src/main/java/org/jenkinsci/plugins/durabletask/ProcessLiveness.java

```diff
@@ -43,6 +54,23 @@
      * @return true if it is apparently still alive (or we cannot tell); false if it is believed to not be running
      */
     public static boolean isAlive(VirtualChannel channel, int pid, Launcher launcher) throws IOException, InterruptedException {
+        Boolean working = workingLaunchers.get(launcher);
+        if (working == null) {
+            // Check to see if our logic correctly reports that an unlikely PID is not running.
+            working = !_isAlive(channel, 9999, launcher);
```

Comment:
The only thing that POSIX guarantees is that pid_t is a signed integral type.  The most portable way to find the size of pid_t, would be to write some C code to get the sizeof pid_t, such as one of the solutions mentioned here: http://stackoverflow.com/questions/12841488/how-can-i-determine-the-maximum-value-of-a-pid-t

Linux has /proc/sys/kernel/pid_max , which works, but is less portable.

JNR has a pid_t interface ( https://github.com/jnr/jnr-ffi/blob/master/src/main/java/jnr/ffi/types/pid_t.java ), but I am not sure if it is possible to get the pid_t size from that.

9999 is not that big, and these days it is possible to hit that.  Although as you correctly mention that the probability of hitting this number is low, it can definitely happen.  I don't like to see these kinds of things, because these kind of booby traps tend to manifest itself when scaling systems like these, and lead to WTF moments.